File: ios/app-obj-c/app-obj-c/ViewController.m
//
//  ViewController.m
//  app-obj-c
//
//  Created by Tim Slot on 23/05/2018.
//  Copyright © 2018 Adversitement. All rights reserved.
//

#import "ViewController.h"
#import "O2MC.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // Init
    self._logTopic = os_log_create("io.o2mc.app-obj-c", "testapp-obj-c");

    // Set endpoint textfield with the currently set endpoint.
    [self.endpointTextField setText:[[O2MC sharedInstance] getEndpoint]];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)BtnTouchCreateEvent:(id)sender {
    os_log(self._logTopic, "created event");

    [[O2MC sharedInstance] track:self.eventNameTextField.text];
}

- (IBAction)BtnTouchResetTracking:(id)sender {
    os_log(self._logTopic, "reset tracking");

    [[O2MC sharedInstance] stop];
    [[O2MC sharedInstance] resume];
}

- (IBAction)BtnTouchStopTracking:(id)sender {
    os_log(self._logTopic, "stop tracking");

    [[O2MC sharedInstance] stop];
}

- (IBAction)InputEndpointChanged:(id)sender; {
    os_log(self._logTopic, "endpoint data changed");

    [[O2MC sharedInstance] setEndpoint:self.endpointTextField.text];
}


@end

File: ios/app-swift/app-swift/ViewController.swift
//
//  ViewController.swift
//  app-swift
//
//  Created by Juan Jose Bernal on 12/07/2018.
//  Copyright © 2018 O2MC. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet weak var eventNameTextField: UITextField!
    @IBOutlet weak var endpointNameTextField: UITextField!
    var _logTopic: OSLog!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if #available(iOS 10.0, *) {
            self._logTopic = OSLog(subsystem: "io.o2mc.app-swift", category: "testapp-swift")
        }
        
        self.endpointNameTextField.text = O2MC.sharedInstance().getEndpoint()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func BtnTouchCreateEvent(_ sender: Any) {
        if #available(iOS 10.0, *) {
            os_log("created event", self._logTopic)
        } else {
            NSLog("created event")
        }
        
        O2MC.sharedInstance().track(self.eventNameTextField.text!)
    }
    
    @IBAction func BtnTouchResetTracking(_ sender: Any) {
        if #available(iOS 10.0, *) {
            os_log("reset tracking", self._logTopic)
        } else {
            NSLog("reset tracking")
        }
        
        O2MC.sharedInstance().stop()
        O2MC.sharedInstance().resume()
    }

    @IBAction func BtnTouchStopTracking(_ sender: Any) {
        if #available(iOS 10.0, *) {
            os_log("stop tracking", self._logTopic)
        } else {
            NSLog("stop tracking")
        }

        O2MC.sharedInstance().stop()
    }

    @IBAction func InputEndpointChanged(_ sender: Any) {
        if #available(iOS 10.0, *) {
            os_log("endpoint changed", self._logTopic)
        } else {
            NSLog("endpoint changed")
        }

        O2MC.sharedInstance().setEndpoint(self.endpointNameTextField.text!)
    }
}

